scipy/spatial/distance.py

```diff
@@ -2975,6 +2990,21 @@ def cdist(XA, XB, metric='euclidean', *, out=None, **kwargs):
         metric_info = _METRIC_ALIAS.get(mstr, None)
         if metric_info is not None:
             cdist_fn = metric_info.cdist_func
+            # TODO: remove out/w checks when distutils
+            # build system is removed
+            if os.name == "nt" and out is not None:
+                required_shape = (mA, mB)
+                if out.shape != required_shape:
+                    raise ValueError("Output array has incorrect shape.")
+                if not out.flags["C_CONTIGUOUS"]:
+                    raise ValueError("Output array must be C-contiguous.")
+                if not np.can_cast(XA.dtype, out.dtype):
+                    raise ValueError("Wrong out dtype.")
+            if os.name == "nt" and "w" in kwargs:
+                w = kwargs["w"]
+                if w is not None:
+                    if (w < 0).sum() > 0:
+                        raise ValueError("Input weights should be all non-negative")
```

Comment:
Since the only difference is `required_shape` maybe factor this out into a function `_extra_windows_error_checks`?